tardis/spectrum/base.py

```diff
@@ -0,0 +1,192 @@
+import warnings
+
+import numpy as np
+from astropy import units as u
+
+from tardis.io.util import HDFWriterMixin
+from tardis.spectrum.formal_integral import IntegrationError
+from tardis.spectrum.spectrum import TARDISSpectrum
+from tardis.util.base import (
+    quantity_linspace,
+)
+
+
+class SpectrumSolver(HDFWriterMixin):
+    hdf_properties = [
+        "montecarlo_virtual_luminosity",
+        "spectrum",
+        "spectrum_virtual",
+        "spectrum_reabsorbed",
+        "spectrum_integrated",
+    ]
+
+    hdf_name = "spectrum"
+
+    def __init__(self, transport_state, spectrum_frequency):
```

Comment:
Consider renaming `spectrum_frequency`

Reply:
perhaps frequency_grid?  I like names that imply a type if we're not going to annotate